### DISCUSSION TITLE
New sampler mentioned in Changelog, how enable?

### DISCUSSION THREAD

**Discussion by brknsoul** [vladmandic/sdnext#1229] [Q&A] [answered]
I noticed that "new sampler: sde++ 2m sde" was mentioned, yet I don't see it in Settings > Sampler Parameters.

**Comment by vladmandic** [comment, marked as answer]
`repositories/k-diffusion` is updated for new installs only.
otherwise, go there and do a `git pull`
also make sure you don't have `k-diffusion` package installed (pip show k-diffusion; pip uninstall k-diffusion) so it actually loads one from repositories.

**Reply by bkosowski** [reply]
Hmmm... Either there is some issue here or I'm not doing something right.

Checking for k-diffusion:

```
$ .\venv\Scripts\activate
$ pip show k-diffusion
WARNING: Package(s) not found: k-diffusion
```

Automatic version when running the webui script:
`17:24:07-228532 INFO     Version: d9f72b06 Wed May 31 09:14:15 2023 -0400`

But when I got to Samplers settings, I see:
![available-samplers](https://github.com/vladmandic/automatic/assets/4819024/3705c686-5e86-415a-9bbd-cb250ac5160e)

**Reply by Reledia** [reply]
@bkosowski  Have you tried doing:
```
cd repositories/k-diffusion 
git pull
```
?

**Reply by bkosowski** [reply]
> @bkosowski Have you tried doing:
> 
> ```
> cd repositories/k-diffusion 
> git pull
> ```
> 
> ?

I didn't even think about it, since I'm running the server with the `--upgrade` flag, which should also upgrade repositories.

OK, but I saw that

```
$ git status
HEAD detached at b43db16
```

which was super weird. I switched to master. And yes, after that, I finally saw the new sampler in the settings.

Thanks for the help.

Still, I think the repository management should address this somehow and update the master branch.

**Reply by vladmandic** [reply]
Installer updates requirements and extensions, it doesn't touch repositories after initial install as they never change - this is the first change ever since I started the fork.

**Reply by bkosowski** [reply]
@vladmandic I see. Yeah, that makes sense. Maybe the `--upgrade` flag should actually attempt to upgrade repositories. Even if there are no upgrades, there wouldn't be much added time to the startup 🤔

**Reply by vladmandic** [reply]
> Maybe the --upgrade flag should actually attempt to upgrade repositories.

yup, i'll add it.
